core/include/detray/propagator/navigator.hpp

```diff
@@ -621,16 +626,79 @@ class navigator {
         if (navigation.trust_level() == navigation::trust_level::e_high or
             navigation.n_candidates() == 1) {
 
+            auto &stepping = propagation._stepping;
+            auto candidate_cache = *navigation.next();
+
             // Update next candidate: If not reachable, 'high trust' is broken
             if (not update_candidate(*navigation.next(), track, det)) {
-                navigation.set_state(navigation::status::e_unknown,
-                                     geometry::barcode{},
-                                     navigation::trust_level::e_no_trust);
-                return;
+
+                // Case 1: When the track overstepped over the overstep
+                // tolerance
+                if (navigation.next()->status ==
+                        intersection::status::e_inside &&
+                    navigation.next()->direction ==
+                        intersection::direction::e_opposite) {
+
+                    const scalar_type new_step_size =
+                        stepping.step_size() / 2.f;
+
+                    // Set unknown if the new step size is smaller than the
+                    // threshold
+                    if (new_step_size < stepping._safety_step_size) {
+
+                        navigation.set_state(
+                            navigation::status::e_unknown, geometry::barcode{},
+                            navigation::trust_level::e_no_trust);
+                        stepping.template release_step<
+                            step::constraint::e_accuracy>();
+                        return;
+                    }
+
+                    stepping.cur_cache = stepping.pre_cache;
+                    stepping
+                        .template set_constraint<step::constraint::e_accuracy>(
+                            new_step_size);
+                    *navigation.next() = std::move(candidate_cache);
```

Comment:
Is this saved next candidate really needed? The navigator should just return here and then automatically go into ```init()``` after the return. That will set the navigation cache to a state that is coherent with the new stepper state again and we can continue to navigate normally, albeit with the step size constraint